src/modules/Tab/TabGroup.tsx

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import {
+  StandardProps,
+  WithStylesProps,
+  ThemeStyleSheetFactory
+} from "../../types";
+import { withStyles } from "../../base";
+
+export interface IProps extends StandardProps<"ul"> {
+  children: JSX.Element[];
+  disabled?: boolean;
+  selectedTab: string;
+  onTabClick: (id: string) => void;
+}
+
+const styleSheet: ThemeStyleSheetFactory = () => ({
+  tabGroup: {
+    display: "flex",
+    margin: "0",
+    padding: "0"
+  }
+});
+
+const TabGroup: React.FC<IProps & WithStylesProps> = ({
+  children,
+  className,
+  cx,
+  disabled,
+  styles,
+  ...props
+}) => {
+  return (
+    <ul className={cx(className, styles.tabGroup)}>
+      {React.Children.map(children, tab =>
+        React.cloneElement(tab, {
+          disabled: tab.props.disabled ?? disabled,
+          ...props
```

Comment:
If we extend the props with `StandardProps<"ul">` then shouldn't they be applied to the `ul` element instead?

Reply:
That's a good question. I see the props as a "mix" of the RadioGroup and the Radio, just because the parent is injecting some props into the children. 
Do you think we shouldn't extend any standard props?

Reply:
I believe it is a good standard to extend the props of the base html element, this can be useful in the future.
I think we should apply the `...props` bit to the `ul` element, and the tabs themselves would extend the `li` props.

Reply:
I agree with Afonso, and I will add my point of view here:

The TabGroup component should be uncontrolled, so it will have its own state. And it can control them by index.

It should have as props: 
- onTabChange;
- onTabClick;
- initialTab (index) , default 0


and the Tab props:
- disabled
- label (string or element in case we want them custom with icons etc)
- id (read optional) (no need to specify as it is a standard prop)


And the TabGroup should have a ref that would allow the user to call
- get: selectedTab()
- set: goNext()
- set: goPrevious()
- set: goTo(index)


optional: could also be listening to url anchor system and #tabtwo or #tabthree would trigger the tab. Could have a hook for that.


do we have animation for changing tabs? it might be good to have them set on diana and we could easily twink that on the ddm then

Reply:
Updated